cygwin needs declaration for sh_setlinebuf

Otherwise the build fails with:
subst.c:7097:7: error: implicit declaration of function 'sh_setlinebuf'; did you mean 'setlinebuf'? [-Wimplicit-function-declaration]
 7097 |       sh_setlinebuf (stdout);
      |       ^~~~~~~~~~~~~
      |       setlinebuf
--- a/subst.c
+++ b/subst.c
@@ -43,6 +43,7 @@
 #include "posixstat.h"
 #include "bashintl.h"
 
+#define NEED_SH_SETLINEBUF_DECL
 #include "shell.h"
 #include "parser.h"
 #include "redir.h"
